vendor/github.com/openshift/library-go/pkg/operator/staticpod/controller/revision/deployment_controller.go

```diff
@@ -106,6 +115,22 @@ func (c RevisionController) createRevisionIfNeeded(operatorSpec *operatorv1.Oper
 		return true, nil
 	}
 
+	if c.validate != nil {
+		if err := c.validate(configs, secrets); err != nil {
```

Comment:
doesn't this validation need to happen *before* we create the configmaps and secrets for the revision?

Reply:
Does not matter much. We have to make sure the config synched to the revisioned variants is correct, all at once and atomically checked. We could read all configmaps and secrets at once, validate them and sync those. But we cannot (form atomicity reasons) validate by reading, then read again to sync. Hence, my change is small, maybe slightly ugly. Valdating before the sync means to change the whe sync logic. Is it worth it?